test/he/test_conn.c

```diff
@@ -1613,3 +1621,53 @@ void test_he_conn_get_curve_name(void) {
   wolfSSL_get_curve_name_ExpectAndReturn(conn.wolf_ssl, curve_name);
   TEST_ASSERT_EQUAL_STRING(curve_name, he_conn_get_curve_name(&conn));
 }
+
+void test_he_conn_start_pmtu_discovery(void) {
+  conn.state = HE_STATE_ONLINE;
+  conn.pmtud_state_change_cb = pmtud_state_change_cb;
+  conn.pmtud_time_cb = pmtud_time_cb;
+
+  he_internal_pmtud_start_base_probing_ExpectAndReturn(&conn, HE_SUCCESS);
+
+  TEST_ASSERT_EQUAL(HE_SUCCESS, he_conn_start_pmtu_discovery(&conn));
+
+  // Do nothing if the pmtud is already started
+  conn.pmtud_state = HE_PMTUD_STATE_BASE;
+  TEST_ASSERT_EQUAL(HE_SUCCESS, he_conn_start_pmtu_discovery(&conn));
+}
+
+void test_he_conn_start_pmtu_discovery_null(void) {
+  TEST_ASSERT_EQUAL(HE_ERR_NULL_POINTER, he_conn_start_pmtu_discovery(NULL));
+}
+
+void test_he_conn_start_pmtu_discovery_invalid_state(void) {
```

Comment:
I would suggest renaming this `test_he_conn_start_pmtu_discovery_fails_when_started_too_early`, so that the test name describes the behavior better instead of simply repeating what the assert does.

Reply:
I think "invalid state" is still a better name. Because calling he_conn_start_pmtu_discovery from any state other than "ONLINE" will return error. It's not about "early" or "late".